datasetinsights/io/tracker/factory.py

```diff
@@ -0,0 +1,115 @@
+import threading
+
+from datasetinsights.constants import TIMESTAMP_SUFFIX
+from datasetinsights.io.tracker.mlflow import MLFlowTracker
+
+
+class TrackerFactory:
+    """Factory: responsible for creating and holding singleton instance
+        of tracker classes"""
+
+    TRACKER = "tracker"
+    CLIENT_ID = "client_id"
+    HOST_ID = "host"
+    EXP_NAME = "experiment"
+    RUN_NAME = "run"
+    MLFLOW_TRACKER = "mlflow"
+    DEFAULT_RUN_NAME = "run_" + TIMESTAMP_SUFFIX
+    __singleton_lock = threading.Lock()
+    __tracker_instance = None
+    RUN_FAILED = "FAILED"
+
+    @staticmethod
+    def create(config, tracker_type):
+        """create tracker class object.
+
+        Args:
+            config : config object, holds server details
+            tracker_type : type of tracker
+
+        Returns:
+            tracker instance.
+        """
+        if TrackerFactory.MLFLOW_TRACKER == tracker_type:
+
+            tracker = config.get(TrackerFactory.TRACKER, None)
+            if tracker and tracker.get(TrackerFactory.MLFLOW_TRACKER, None):
+                mlflow_config = tracker.get(TrackerFactory.MLFLOW_TRACKER)
```

Comment:
This is calling for some validation logic against the confit yaml file of the mlflow tracker section. We should provide enough error logging if the config is specified correctly. Otherwise, user will see mysterious error with no clues how to fix the yaml file.